### PR TITLE
Make LTI "Custom role" configuration more configurable

### PR DESCRIPTION
**Pull request by Lyx52**
Hi, this is my first contribution to this repository.

* The problem:
Current implementation of custom roles allows only one custom role also it only accepts string that are exactly equal to role name which can limit configurability. 

This pull request implements ability to define multiple custom roles and custom roles that need to be added for example:
```
lti.custom_role_name.1=Instructor
lti.custom_roles.1=ROLE_MOODLE_INSTRUCTOR

lti.custom_role_name.2=Learner
lti.custom_roles.2=ROLE_MOODLE_LEARNER

-- Additionally theres regex support for matching roles --
lti.custom_role_name.3=ims\/lis\/Administrator
lti.custom_roles.3=ROLE_MOODLE_ADMINISTRATOR
```

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

File: modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java [hunk 1]
```diff
@@ -253,10 +260,29 @@ protected void activate(ComponentContext cc) {
...
+    // Add support for legacy custom_role_name
+    String customRoleName = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLE_NAME));
+
     if (customRoleName != null) {
-      String custumRolesString = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLES));
-      customRoles = custumRolesString.split(",");
+      String customRolesString = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLES));
+      String[] customRoles = customRolesString.split(",");
+      Pattern customRolePattern = Pattern.compile(customRoleName);
+      customRolePatterns.put(customRolePattern, customRoles);
```
Comment (Arnei) [inline modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java]: Little bit of code duplication going on here, same code can be found in lines 273-275. Might be worth turning this into a function?

**Review by Arnei — CHANGES_REQUESTED** [review]
Hi Lyx52,

thanks for contributing and sorry it took a bit to get a review for this.

I'm not knowledgable on LTI so I can't comment on whether this change makes sense, but the code looks good to me and does what it says.

Please add the new config keys and an explanation to the relevant configuration file, so others can know about this feature too.

I also think this would be worth mentioning in the release notes for Opencast 15, so please add a short note in `/docs/guides/admin/docs/releasenotes`.

File: modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java (hunk 1 above)
Comment (Lyx52) [reply modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java]: Hi sorry for not answering sooner didnt check my github activity, one question about release notes I see that there's no Opencast 15 version heading, should I add one?

Comment (Arnei) [reply modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java]: No, you don't need to. Just add your own `changes-to-lti-configuration.txt` to the folder, with a short note on what changed. The release managers will take care of proper formatting when OC15 releases.

**Review by Arnei — APPROVED** [review]

**Comment by Arnei** [comment]
Seeing as the ICLA has been signed for ... quite some time now, I believe we can go ahead and merge this. Sorry for not getting this into OC 15.

**Pull request merged**